service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/FishingActivityMapper.java

```diff
@@ -536,15 +534,19 @@ private void getQuantityFromAapProduct(FaCatchEntity faCatch, Double quantity) {
 
             SizeDistribution specifiedSizeDistribution = faCatch.getSpecifiedSizeDistribution();
             if (specifiedSizeDistribution != null) {
-                SizeDistributionEntity sizeDistributionEntity = SizeDistributionMapper.INSTANCE.mapToSizeDistributionEntity(specifiedSizeDistribution);
-                sizeDistributionEntity.setFaCatch(faCatchEntity);
-                List<CodeType> classCodes = specifiedSizeDistribution.getClassCodes();
-                for (CodeType classCode : Utils.safeIterable(classCodes)) {
-                    SizeDistributionClassCodeEntity sizeDistributionClassCodeEntity = SizeDistributionMapper.INSTANCE.mapToSizeDistributionClassCodeEntity(classCode);
-                    sizeDistributionEntity.addSizeDistribution(sizeDistributionClassCodeEntity);
+                if (specifiedSizeDistribution.getCategoryCode() != null) {
+                    faCatchEntity.setSizeDistributionCategoryCode(specifiedSizeDistribution.getCategoryCode().getValue());
+                    faCatchEntity.setSizeDistributionCategoryCodeListId(specifiedSizeDistribution.getCategoryCode().getListID());
                 }
 
-                faCatchEntity.setSizeDistribution(sizeDistributionEntity);
+                List<CodeType> classCodes = specifiedSizeDistribution.getClassCodes();
+                if (!CollectionUtils.isEmpty(classCodes)) {
+                    if (classCodes.size() > 1) {
+                        throw new IllegalArgumentException("Failed to map FACatch.specifiedSizeDistribution, more than one SizeDistributionType.ClassCode found");
```

Comment:
Ah yes, this is it! I think it is very helpful if the values from the involved classCodes are provided in the description.